### PR TITLE
Process static data

### PR DESCRIPTION
**Pull request by JoelClemence**

### CODE REVIEW

File: deployment/src/main/terraform/main.tf [hunk 1]
```diff
@@ -76,8 +77,10 @@ module functions {
   new_partitioned_raw_queue_url        = module.messaging.new_partitioned_file_queue_url
   trigger_resample_jar                 = var.TRIGGER_RESAMPLE_FUNCTION_LAMBDA_JAR_PATH
   resampling_spark_job_jar_name        = var.RESAMPLING_SPARK_JOB_JAR_NAME
-  ais_static_data_store_name           = data.external.secrets.result["static_data_store"]
+  ais_static_data_store_name           = data.external.secrets.result["static_data_upload_store"]
```
Comment (agraham1988) [inline deployment/src/main/terraform/main.tf]: wrong data store

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,18 +1,21 @@
 resources:
```
Comment (oliverCausley) [inline azure-pipelines.yml]: Do we need to think about how this clashes with the 'Add POOL to YAML' changeset?

File: deployment/src/main/terraform/main.tf (hunk 1 above)
Comment (oliverCausley) [inline deployment/src/main/terraform/main.tf]: should the names be consistent with those found in the storage module?

Comment (oliverCausley) [reply deployment/src/main/terraform/main.tf]: they seem to be referred to as following in different modules
e.g
 ais_static_data_store_name or static_data_store_name 
 ais_static_data_upload_store_name or static_data_upload_store_name

File: lambdas/process-static-data-file/src/test/python/test_lambda_handler.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        try:
+            lambda_handler(request, {})
+            self.fail("Was expecting lambda handler to raise exception")
+        except Exception as e:
+            self.assertEqual(str(e), expected_error_message)
+        finally:
+            mock_new_file_handler.assert_not_called()
+
+    @patch("process_new_static_file.new_static_file_handler." "NewFileHandler")
+    def testWhenMultipleRecordsThenRaisesException(self, mock_new_file_handler):
```
Comment (oliverCausley) [inline lambdas/process-static-data-file/src/test/python/test_lambda_handler.py]: Should testWhenMultipleRecordsThenRaisesException be testWhenCalledWithMultipleRecordsThenRaisesException?

Comment (oliverCausley) [inline lambdas/process-static-data-file/src/test/python/test_lambda_handler.py]:
```suggestion
    def testWhenCalledWithMultipleRecordsThenRaisesException(self, mock_new_file_handler):
```

File: lambdas/process-static-data-file/src/test/python/test_lambda_handler.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        self.assertEqual({}, response)
+        mock_handler.handle_new_file.assert_called_with(
+            expected_input_file, expected_destination
+        )
+        mock_handler.with_buffer_size.assert_called_with(1024)
+        mock_handler.with_output_file_extension.assert_called_with(".csv")
+
+    @patch("process_new_static_file.new_static_file_handler.NewFileHandler")
+    def testWhenCalledWithNoFilesThenRaisesError(self, mock_new_file_handler):
```
Comment (oliverCausley) [inline lambdas/process-static-data-file/src/test/python/test_lambda_handler.py]: Optional: Error could be Exception?

File: azure-pipelines.yml (hunk 2 above)
Comment (JoelClemence) [reply azure-pipelines.yml]: Probably - but we can merge properly

**Comment by oliverCausley** [comment]
Do we need to up the version number?

**Comment by JoelClemence** [comment]
requires #81 to be merged into master

File: buildSrc/src/main/kotlin/uk/gov/ukho/gradle/plugin/python/PythonPlugin.kt [hunk 5]
```diff
@@ -66,6 +66,11 @@ class PythonPlugin : Plugin<Project> {
         tasks.register<SystemPythonTask>(PythonTask.virtualEnvSetup) {
             dependsOn(PythonTask.virtualEnvInstall)
             args("virtualenv", pythonVirtualEnvDir)
+            doLast {
+                exec {
+                    commandLine("chmod", "-R", "+rwx", pythonVirtualEnvDir)
```
Comment (galexite) [inline buildSrc/src/main/kotlin/uk/gov/ukho/gradle/plugin/python/PythonPlugin.kt]: I don't think this is necessary? Also, shouldn't this be applied only the `bin` directory?

File: deployment/src/main/terraform/modules/data-query/processed_static_data_table.tf [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    columns {
+      name = "vessel_type_index"
+      type = "double"
+    }
+
+    columns {
+      name = "message_type_id"
+      type = "int"
+    }
+
```
Comment (ciaranevans) [inline deployment/src/main/terraform/modules/data-query/processed_static_data_table.tf]: Inconsistent formatting in terms of lines between the columns

File: buildSrc/src/main/kotlin/uk/gov/ukho/gradle/plugin/python/PythonPlugin.kt (hunk 5 above)
Comment (JoelClemence) [reply buildSrc/src/main/kotlin/uk/gov/ukho/gradle/plugin/python/PythonPlugin.kt]: Yes this is necessary for the activate script to work but also for clean to work. At the end of the day these are built files

File: lambdas/process_new_static_file/src/test/python/test_lambda_handler.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+        try:
+            lambda_handler(request, {})
+            self.fail("Was expecting lambda handler to raise exception")
+        except Exception as e:
+            self.assertEqual(str(e), expected_error_message)
+        finally:
+            mock_new_file_handler.assert_not_called()
+
+    @patch("process_new_static_file.new_static_file_handler." "NewFileHandler")
```
Comment (eganjs) [inline lambdas/process_new_static_file/src/test/python/test_lambda_handler.py]: Space here? is this valid?

Comment (JoelClemence) [reply lambdas/process_new_static_file/src/test/python/test_lambda_handler.py]: Still valid - will fix separately

**Pull request merged**